ocaml/sdk-gen/go/autogen/src/jsonrpc_client.go

```diff
@@ -0,0 +1,225 @@
+		return nil, fmt.Errorf("could not create request for %v() : %w", request.Method, err)
+	}
+
+	httpResponse, err := client.httpClient.Do(httpRequest)
+	if err != nil {
+		return nil, fmt.Errorf("call %v() on %v. Error making http request: %w", request.Method, httpRequest.URL.Redacted(), err)
+	}
+	defer httpResponse.Body.Close()
+
+	var rpcResponse *Response
+	body, err := io.ReadAll(httpResponse.Body)
+	if err != nil {
+		return nil, fmt.Errorf("call %v() on %v status code: %v. Could not read response body: %w", request.Method, httpRequest.URL.Redacted(), httpResponse.StatusCode, err)
+	}
+	body = convertUnhandledJSONData(body)
+	err = json.Unmarshal(body, &rpcResponse)
+	if err != nil && !errors.Is(err, io.EOF) {
+		return nil, fmt.Errorf("call %v() on %v status code: %v. Could not decode response body: %w", request.Method, httpRequest.URL.Redacted(), httpResponse.StatusCode, err)
+	}
+
+	if rpcResponse == nil {
+		return nil, fmt.Errorf("call %v() on %v status code: %v. Response missing", request.Method, httpRequest.URL.Redacted(), httpResponse.StatusCode)
+	}
+
+	return rpcResponse, nil
+}
+
+func (client *rpcClient) sendCall(methodName string, params ...interface{}) (result interface{}, err error) {
+	response, err := client.call(context.Background(), methodName, params...)
+	if err != nil {
+		return
+	}
+
+	if response.Error != nil {
+		errString := fmt.Sprintf("API error: code %d, message %s", response.Error.Code, response.Error.Message)
+		if response.Error.Data != nil {
+			errString += fmt.Sprintf(", data %v", response.Error.Data)
+		}
+		err = errors.New(errString)
+		return
+	}
+
+	result = response.Result
+	return
+}
+
+type SecureOpts struct {
+	ServerCert string
+	ClientCert string
+	ClientKey  string
+}
+
+type ClientOpts struct {
+	URL        string
+	SecureOpts *SecureOpts
+	Timeout    int
+	Headers    map[string]string
+}
+
+func newJSONRPCClient(opts *ClientOpts) *rpcClient {
+	client := &rpcClient{
+		endpoint:   fmt.Sprintf("%s%s", opts.URL, "/jsonrpc"),
+		httpClient: &http.Client{},
+		headers:    make(map[string]string),
+	}
+
+	if strings.HasPrefix(opts.URL, "https://") {
+		skipVerify := true
+		caCertPool := x509.NewCertPool()
+		certs := []tls.Certificate{}
+		if opts.SecureOpts != nil {
+			skipVerify = false
+			if opts.SecureOpts.ServerCert != "" {
+				caCert, err := os.ReadFile(opts.SecureOpts.ServerCert)
+				if err != nil {
+					log.Fatal(err)
+				}
+				caCertPool.AppendCertsFromPEM(caCert)
+			}
+			if opts.SecureOpts.ClientCert != "" || opts.SecureOpts.ClientKey != "" {
+				if opts.SecureOpts.ClientCert == "" {
+					log.Fatal(errors.New("missing client certificate"))
+				}
+				if opts.SecureOpts.ClientKey == "" {
+					log.Fatal(errors.New("missing client private key"))
+				}
+				cert, err := tls.LoadX509KeyPair(opts.SecureOpts.ClientCert, opts.SecureOpts.ClientKey)
+				if err != nil {
+					log.Fatal(err)
+				}
+				certs = []tls.Certificate{cert}
+			}
+		}
+		tlsConfig := &tls.Config{
```

Comment:
In the `tls.Config`, the `CipherSuites` needs to be set explicitly for sake of security.